opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java

```diff
@@ -164,4 +177,208 @@ public void testEventOnCreationAndDeletion() throws Exception {
+    }
+    @Test
+    @Transactional
+    public void testUnlimitedLocations10() throws Exception {
+        // 10 <= unlimited, should 10
+        assertEquals(10, testLocationLimit(10,UNLIMITED_FLAG));
+    }
+    @Test
+    @Transactional
+    public void testUnlimitedLocations15() throws Exception {
+        // 15 <= unlimited, should return 15
+        assertEquals(15, testLocationLimit(15,UNLIMITED_FLAG));
+    }
+    @Test
+    @Transactional
+    public void testLimit5Locations5() throws Exception {
+        // 5 <= specified limit of 5, should return 5
+        assertEquals(5, testLocationLimit(5,5));
+    }
+
+    @Test
+    @Transactional
+    public void testLimit5Locations10() throws Exception {
+        // 10 > specified limit of 5, should return 5
+        assertEquals(5, testLocationLimit(10,5));
+    }
+    @Test
+    @Transactional
+    public void testLimit5Locations15() throws Exception {
+        // 15 > specified limit of 5, should return 5
+        assertEquals(5, testLocationLimit(15,5));
+    }
+    @Test
+    @Transactional
+    public void testLimit10Locations5() throws Exception {
+        // 5 <= specified limit of 10, should return 6 (includes default)
+        assertEquals(5, testLocationLimit(5,10));
+    }
+
+    @Test
+    @Transactional
+    public void testLimit10Locations10() throws Exception {
+        // 10 = specified limit of 10, should return 10
+        assertEquals(10, testLocationLimit(10,10));
+    }
+
+    @Test
+    @Transactional
+    public void testLimit10Locations15() throws Exception {
+        // 10 > specified limit of 10, should return 10
+        assertEquals(10, testLocationLimit(15,10));
+    }
+
+    @Test
+    @Transactional
+    public void testLimit15Locations5() throws Exception {
+        // 5 <= 15 specified limit of 15, should return 5 (includes default)
+        assertEquals(5, testLocationLimit(5,15));
+    }
+    @Test
+    @Transactional
+    public void testLimit15Locations10() throws Exception {
+        // 10 <= specified limit of 15, should return 10
+        assertEquals(10, testLocationLimit(10,15));
+    }
+    @Test
+    @Transactional
+    public void testLimit15Locations15() throws Exception {
+        // 10 > specified limit of 15, should return 15
+        assertEquals(15, testLocationLimit(15,15));
+    }
+
+    public static Integer calcLimit(final Integer flaggedLimit){
+        if (flaggedLimit == DEFAULT_FLAG) {
+            return DEFAULT_LIMIT;
+        } else if (flaggedLimit.equals(UNLIMITED_FLAG)){
+            return UNLIMITED;
+        } else {
+            return flaggedLimit;
+        }
+    }
+    /**
+     * Add LOCATION_COUNT locations
+     * Store them locally as we add them
+     * Then fetch the list and compare
+     */
+    private int testLocationLimit(final Integer LOCATION_COUNT, final Integer LIMIT) throws Exception {
+        ArrayList<Location> localLocations = new ArrayList<>(calcLimit(LOCATION_COUNT));
```

Comment:
Why do you initialize the array with a size? We only need something > 10, so 11 or 12 is sufficient. To initialize an array with Integer.MAX_VALUE does not make any sense.